src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -105,18 +113,19 @@ public ReadsSparkSource(final JavaSparkContext ctx, final ValidationStringency v
                     readFileName, AnySAMInputFormat.class, LongWritable.class, SAMRecordWritable.class,
                     conf);
 
-        return rdd2.map(v1 -> {
+        JavaRDD<GATKRead> reads= rdd2.map(v1 -> {
             SAMRecord sam = v1._2().get();
             if (isBam || samRecordOverlaps(sam, intervals)) { // don't check overlaps for BAM since it is done by input format
                 return (GATKRead) SAMRecordToGATKReadAdapter.headerlessReadAdapter(sam);
             }
             return null;
         }).filter(v1 -> v1 != null);
+        return putPairsInSamePartition(header, reads);
     }
 
     /**
      * Loads Reads using Hadoop-BAM. For local files, readFileName must have the fully-qualified path,
-     * i.e., file:///path/to/bam.bam. This excludes unmapped reads.
+     * i.e., file:///path/to/bam.bam.
```

Comment:
Yes, previously it used `IntervalUtils. getAllIntervalsForReference()` which excluded unmapped reads, and which is inconsistent with the non-Spark version. Unmapped reads should be filtered out if they are not needed. Note that in `GATKSparkTool.getReads()` it says "If no intervals were specified, returns all the reads (both mapped and unmapped).".